src/components/RangeInput/RangeInput.js

```diff
@@ -0,0 +1,111 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import autoHideContainerHOC from '../../decorators/autoHideContainer.js';
+import headerFooterHOC from '../../decorators/headerFooter.js';
+
+export class RawRangeInput extends Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      min: props.values.min,
+      max: props.values.max,
+    };
+  }
+
+  componentWillReceiveProps(nextProps) {
+    this.setState({
+      min: nextProps.values.min,
+      max: nextProps.values.max,
+    });
+  }
+
+  onChange = name => event => {
+    this.setState({
+      [name]: event.currentTarget.value,
+    });
+  };
+
+  onSubmit = event => {
+    event.preventDefault();
+
+    this.props.refine([this.state.min, this.state.max]);
+  };
+
+  get isDisabled() {
+    return this.props.min >= this.props.max;
+  }
+
+  render() {
+    const { min: minValue, max: maxValue } = this.state;
+    const { min, max, step, cssClasses, labels } = this.props;
+
+    return (
+      <form className={cssClasses.form} onSubmit={this.onSubmit}>
+        <fieldset className={cssClasses.fieldset}>
+          <label className={cssClasses.labelMin}>
+            <input
+              className={cssClasses.inputMin}
+              type="number"
+              min={min}
```

Comment:
I think we should validate it, because the `min` & `max` are the range where the user is able to search. It doesn't make sense to let it put an higher or lower. IMO we need to show a hint (in this case the `HTML5` validation) to the user that his search is out of bounds and so irrelevant.

With the actual implementation one thing that it might happen if we don't set the limit on the input is this:

* the user set the input out of bounds
* the user submit the form
* the search will be trigger with original values
* the render method will erase your current out of bounds values and replaced with the actual state

One thing that we could do is to be more precise when we refine. Currently, we just check that [one of the values has changed](https://github.com/algolia/instantsearch.js/blob/develop/src/connectors/range-slider/connectRangeSlider.js#L134) if its the case we clear the refinements. Then we set the refinement **IF** [the value are correct](https://github.com/algolia/instantsearch.js/blob/develop/src/connectors/range-slider/connectRangeSlider.js#L142). But the search & render will be trigger [even if the value is not set](https://github.com/algolia/instantsearch.js/blob/develop/src/connectors/range-slider/connectRangeSlider.js#L162). A solution could be to handle the values separately, in this way we could pass a falsy value without triggering the search & render.

WDYT?